packages/molecule/src/resolve.ts

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env node
+import { ParseResult } from "./type";
+import { codegenReturnWithElements } from "./codegen";
+import { Config } from "./config";
+import * as fs from "node:fs";
+import * as path from "path";
+
+function camelcase(str: string): string {
+  return str.replace(/_([a-z])/g, (_, letter) => letter.toUpperCase());
+}
+
+export type RelativePath = string;
+export type FileWithDependence = {
+  relativePath: string;
+  dependencies: string[];
+};
+
+export function resolveDependencies(
+  importPath: RelativePath,
+  baseDir: string,
+  resolved: Set<RelativePath>
+): FileWithDependence[] {
+  const dependencies: FileWithDependence[] = [];
+  // check if the file exist
+  const realPath = path.join(baseDir, importPath);
+  if (!fs.existsSync(realPath)) {
+    console.error(`Schema file ${realPath} does not exist.`);
+    process.exit(1);
+  }
+
+  const cur: FileWithDependence = {
+    relativePath: importPath,
+    dependencies: [],
+  };
+
+  const schema = fs.readFileSync(realPath, "utf-8");
+  if (!schema) {
+    return [cur];
+  }
+
+  const matched = schema.match(/.*import\s+"(.*)".*;/g);
+  if (!matched) {
+    return [cur];
+  }
+
+  // collect all import filenames
+  const importFileNames = matched
+    .map((item: string) => {
+      // if is comment statement, continue
+      if (item.trim().startsWith("//")) {
+        return "";
+      }
+      const m = item.match(/.*"(.*)".*/);
+      return m ? m[1] : "";
+    })
+    .filter(Boolean);
+
+  // loop all import files
+  for (const importFileName of importFileNames) {
+    const mFilePath = path.join(baseDir, importFileName + ".mol");
+    const mRelativePath = path.relative(baseDir, mFilePath);
+
+    cur.dependencies.push(importFileName);
+    if (!resolved.has(mFilePath)) {
+      // mask this file has resolved
+      resolved.add(mFilePath);
+
+      const _dependencies = resolveDependencies(
+        mRelativePath,
+        baseDir,
+        resolved
+      );
+      dependencies.push(..._dependencies);
```

Comment:
```suggestion
      dependencies.push(...resolveDependencies(
        mRelativePath,
        baseDir,
        resolved
      ));
```

It's better to avoid using variables that start with an underline as they often imply that the variable is private and mutable in a scope.

Reply:
I will avoid this way of writing in the future